sys/mips/mips/pmap.c

```diff
@@ -2036,6 +2042,14 @@ pmap_protect(pmap_t pmap, vm_offset_t sva, vm_offset_t eva, vm_prot_t prot)
 					va = va_next;
 				}
 			}
+			if (pte_test(&pbits, PTE_SC) == 0) {
```

Comment:
Can this all be moved under PTE_D? Currently it feels odd to be doing this *after* messing with `va` (and potentially calling `pmap_invalidate_range`)...

Reply:
Good call.

Reply:
I've changed a few other points where both PTE_D and PTE_SC were being manipulated; there's no point in setting PTE_SC if we just cleared PTE_D, as we can leave the page capdirty but still catch writes.  The MI bits still get updated at these points, and so if anyone up there cares, they can call back down to make the page cap-clean; I don't think anyone cares like that.